src/types.rs

```diff
@@ -303,34 +338,48 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
     }
   }
 
-  pub fn get_name(&self) -> NameType {
-    self.name.clone()
+  pub fn get_name<'a>(&'a self) -> &'a NameType {
```

Comment:
It was initially doing implicit copy and then you and Peter both agreed on changing it (see +1 on your previous comment). 
Its not that clean either way actually. Can you guys give me a final comment saying what is the final call on this? I am happy to revert it back to implicit copy. @inetic @benjaminbollen